dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java

```diff
@@ -193,13 +320,75 @@ public void shouldDeleteRules() throws DotSecurityException, DotDataException {
         final Host host = new SiteDataGen().nextPersisted();
         final Rule rule = new RuleDataGen().host(host).nextPersisted();
 
-        this.addPermission(role, host, true);
+        this.addRulesPermission(role, host, true);
+
+        assertTrue(existRule(rule));
+
+        rulesAPI.deleteRule(rule, user, false);
+
+        assertFalse(existRule(rule));
+    }
+
+    /**
+     * Method to Test: {@link RulesAPIImpl#deleteRule(Rule, User, boolean)}
+     * When: User with permission try to delete a rule's page
+     * Should: Delete it
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldDeleteRulesInPage() throws DotSecurityException, DotDataException {
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role).nextPersisted();
+        final Host host = new SiteDataGen().nextPersisted();
+
+        final Template template = new TemplateDataGen().nextPersisted();
+        final HTMLPageAsset htmlPageAsset = new HTMLPageDataGen(host, template)
+                .nextPersisted();
+
+        final Rule rule = new RuleDataGen().page(htmlPageAsset).nextPersisted();
 
-        assertTrue(existRule(user, rule));
+        this.addRulesPermission(role, host, true);
+        this.addPermission(role, htmlPageAsset, PermissionAPI.PERMISSION_PUBLISH);
+
+        assertTrue(existRule(rule));
+
+        rulesAPI.deleteRule(rule, user, false);
+
+        assertFalse(existRule(rule));
+
+    }
+
+    /**
+     * Method to Test: {@link RulesAPIImpl#deleteRule(Rule, User, boolean)}
+     * When: User with permission try to delete a rule's page without PUBLISH permission over the page
+     * Should:throw a {@link DotSecurityException}
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test(expected = DotSecurityException.class)
+    public void shouldDeleteRulesInPageShouldNotWork() throws DotSecurityException, DotDataException {
```

Comment:
Rename method to: `shouldNotDeleteRuleOnPage`

Reply:
done https://github.com/dotCMS/core/pull/18201/commits/a3d5384f514af695b89fdbc549338def9a8cce2c#diff-9baded4fb0fb63e2648d9c9c4380ec1aR385